### PR TITLE
Remove `querystring_parser`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/12960?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12960/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12960
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

See #12953 for context. This PR removes `querystring_parser`.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/server/handlers.py [hunk 1]
```diff
@@ -477,16 +477,8 @@ def _get_request_json(flask_request=request):
...
 
 def _get_request_message(request_message, flask_request=request, schema=None):
-    from querystring_parser import parser
-
     if flask_request.method == "GET" and len(flask_request.query_string) > 0:
-        # This is a hack to make arrays of length 1 work with the parser.
-        # for example experiment_ids%5B%5D=0 should be parsed to {experiment_ids: [0]}
-        # but it gets parsed to {experiment_ids: 0}
-        # but it doesn't. However, experiment_ids%5B0%5D=0 will get parsed to the right
-        # result.
```
Comment (harupy) [inline mlflow/server/handlers.py:487]: Need to check whether MLflow UI sends requests with this format.

Comment (daniellok-db) [reply mlflow/server/handlers.py:487]: Hmm, one weird example is for the `get-history-bulk-interval` API (fetching metrics for charts)

The query string looks like:
```
http://localhost:3000/ajax-api/2.0/mlflow/metrics/get-history-bulk-interval?run_ids=2a1a38961b4c4bb684dfd084451daeab&run_ids=9c6f757ffdaa42afafb948d14d191757&metric_key=precision&max_results=320
```

Comment (harupy) [reply mlflow/server/handlers.py:487]: `get-history-bulk-interval` is already using `args.to_dict(flat=False)`:

https://github.com/mlflow/mlflow/blob/0c26482f023078d723aa13cc604ace7335e8b9a1/mlflow/server/handlers.py#L1103-L1106

**Comment by harupy** [comment]
@daniellok-db For the experiment select feature, the `experiments` parameter is parsed on the frontend and translated to `experiment_ids`:

image

**Review by daniellok-db — APPROVED** [review]
Overall LGTM. I had a quick look at the `service.proto` file and it looks like `GetMetricHistoryBulkInterval` (which it looks like it works fine) and `SearchTraces` (which is broken even before this PR due to comma separated string) are the only endpoints with array type arguments.

**Pull request merged**